README.md

```diff
@@ -1,11 +1,87 @@
 
 A rough state diagram of the contract looks like this:
 
 ![dx-mgn-pool state diagram](dx-mgn-pool%20state%20machine.png)
 
+The deployment consists of two smart contracts, `DxMgnPool` and a `Coordinator`
+
+### DxMgnPool
+
+This contract allows participants to deposit a single predefined ERC20 token (e.g. GNO) into the contract's balance. 
+The contract will continuously participate with its entire balance in dutchX auctions and thus generate magnolia.
+We deploy two instances of the same contract: one allowing to deposit a token (e.g. GNO) and another one allowing to deposit another token (e.g. WETH). 
+The two contracts will continuously trade their deposit token against the deposit token of the other pool. 
+
+#### Auction process
+1. Initially, the pool participates in the auction *depositToken* -> *secondaryToken*. 
+It does so by posting a sell order for the full balance of its deposit token. 
+2. After the first auction finishes, the contract claims all of its proclaims (in *secondaryToken*) and reinvests it as a sell order in the reverse auction (*secondaryToken* -> *depositToken*). 
+3. After the reverse auction finishes, it claims all of its proclaims in the original deposit token and starts over.
+
+One such round will take two dutchX auctions to finish, thus roughly 12 hours.
+The pooling period si defined to end after an even number of auctions. 
+Thus, participants can eventually withdraw the same ERC20 token they deposited.
+
+With two deployed instances and our example above, the first pool trades GNO for WETH (step 1) followed by WETH for GNO (step 2).
+The second pool starts by trading WETH for GNO followed by GNO for WETH.
+
+#### Deposits
+Deposits can happen at any time during the pooling period. 
+A participant can post multiple deposits at different times. 
+The contract computes the value of the deposit compared to the total balance it currently holds and eventually uses this "share" - combined with the time of deposit - to calculate the share of MGN that each participation contributed (see below).
+Note, that a deposit only starts contributing to the MGN generation in the next even auction index (since the contract can only invest it in the *depositToken -> secondaryToken* auction)
+
+#### Deposit Withdraws
+Participants can withdraw their deposit **only after the pooling period has ended**.
+Note, that due to price fluctuations it is possible that people withdraw a different amount than they deposited. 
+E.g. if they intially provided 10% of the pool and the price of GNO fell over the time of pooling, the contract will hold less GNO at the end.
+Particpants are only entitled to withdraw their share, which would then be worth less than the initial deposit.
+
+Note, that in order to avoid adjusting all shares after each deposit, we don't store the relative contribution of each participation in %, but instead store an absolute number of *pool shares* (more gas efficient).
```

Comment:
```suggestion
Note that, in order to avoid adjusting all shares after each deposit, we don't store the relative contribution of each participation as a percentage, but rather as an absolute number of *pool shares* (more gas efficient).
```